cadence/contracts/services/InterchainTokenService.cdc

```diff
@@ -0,0 +1,324 @@
+        if (metadataVersion > self.LATEST_METADATA_VERSION) {
+            panic("Invalid Metadata Version")
+        }
+
+        let datahash = Crypto.hash(metadata, algorithm: HashAlgorithm.SHA3_256)
+
+        emit InterchainTransfer(
+            tokenAddress: contractAddress,
+            tokenName: contractName,
+            sourceAddress: senderIdentity.address,
+            destinationChain: destinationChain,
+            amount: amount,
+            datahash: datahash
+        )
+
+        //Awaiting ABI.Encode/decode support to parse metadata
+
+        AxelarGateway.callContract(
+            senderIdentity: senderIdentity,
+            destinationChain: destinationChain,
+            destinationContractAddress: destinationAddress,
+            payload: metadata
+        )
+    }
+
+    access(self) fun _processCommand(commandSelector: [UInt8], payload: [UInt8]){
+        //call function based on commandSelector
+    }
+
+    access(self) fun _withdraw(amount: UFix64, tokenAddress: Address, tokenName: String, reciever: &{FungibleToken.Receiver}){
+        if (amount > 0.0){
+            panic("Amount must be greater than zero")
+        }
+        if(!self.tokens.keys.contains(tokenAddress)){
+            panic("Token not onboarded")
+        }
+        if (self.tokens[tokenAddress] == TokenManagerType.Native){
+            // create nativeToken reference
+            let nativeToken = (&self.nativeTokens[tokenAddress] as &NativeTokens?) ?? panic("could not borrow native token ref")
+            
+            //borrow the coresponding native token vault
+            let vault <- self.account.load<@FungibleToken.Vault>(from: nativeToken.vaultRef)!
+
+            //withdraw from vault and deposit to reciever
+            let tempVault <- vault.withdraw(amount: amount)
+            reciever.deposit(from: <-tempVault)
+
+            //save vault back to storage
+            self.account.save(<- vault, to: nativeToken.vaultRef)
+        } else {
+            let managedToken = (&self.managedTokens[tokenAddress] as &ManagedTokens?) ?? panic("could not borrow managed token ref")
+            let mintCap <- self.account.load<@AxelarFungibleTokenInterface.Minter>(from: managedToken.minterCapability)!
+            let mintVault <- mintCap.mintTokens(amount: amount)!
+            reciever.deposit(from: <-mintVault)
+            self.account.save(<- mintCap, to: managedToken.minterCapability)
+        }
+   }
+
+    access(self) fun _launchToken(name: String, symbol: String): Address{
+        let tokenAccount = AuthAccount(payer: self.account)
+        let tokenAccountAddress = tokenAccount.address
+        let key = PublicKey(
+            publicKey: self.tokenPublicKey.decodeHex(),
+            signatureAlgorithm: SignatureAlgorithm.ECDSA_P256
+        )
+        tokenAccount.keys.add(
+            publicKey: key,
+            hashAlgorithm: HashAlgorithm.SHA3_256,
+            weight: 10.0
```

Comment:
Why is a key with 1% signing auth added to the token account? Also, is there intention to update this for multisig?

On the topic of signing authority, an important note on this distributed deployment pattern - all of the contracts deployed by the ITS will need to be updated due to breaking updates come Cadence 1.0. The parties with threshold signing auth (>= 1000.0) will need to execute updates on all deployed contracts via multisig or the updates will need to be staged beforehand and executed by someone/system after the 1.0 milestone.

More info on that last point here: https://forum.flow.com/t/cadence-1-0-upgrade-plan/5477

Regardless of update plans, `tokenAccount` will need to have keys totaling 1000.0 weight within this function's scope to perform any future signed actions with those accounts.

Reply:
I updated this to be 1000.0, thanks for clarifying that.  

I am not sure if there will be any multisig functionality in the token accounts in the future.  This will depend on what axelar implements on other chains for token bridging.

I will reach out to milap to clarify.